### ISSUE TITLE
Allow limiting the `MOLD_ELF_TARGETS` CMake variable to improve build times.

### ISSUE THREAD

**Issue by no-more-secrets** [rui314/mold#1364]
For my purposes I only need `MOLD_ELF_TARGETS=X86_64` which I suspect would significantly increase build times, but the CMake files don't seem to allow a way to override it.  Perhaps you should only fill it with values if it is not already set.

**Comment by rui314** [comment]
That feature is intentionally left out because if we provide a configurable CMake variable for it, it is not hard to imagine that someone would use it to build an x86-64-only mold binary and distribute it widely. We want all mold binaries to be able to be used as a cross linker, so we don't want to encourage that situation.

See https://github.com/rui314/mold/blob/main/CMakeLists.txt#L38-L44

**Comment by rui314** [comment]
Right. It's technically doable, and I wouldn't stop you from doing it if you know what you are doing. It's open source, and you can make whatever changes you want. I just don't want to provide a built-in switch to turn all non-native targets off.

**Comment by to-audiobook** [comment]
@rui314 
first of all. Thanks for `mold`. The linker performance really is impressive. Second, `mold` is yours and you can do whatever you think it is best for it. 

But I really think you should reconsider this decision. I don't believe the rationale behind it is in anyway enforceable. Anyone using a pre-built binary found in the wild would be foolish to assume anything about it. 

Also, and this is the most important part, having just a single target makes the build 1000x faster and, as @absolutelynothinghere showed, the project is structured in such a clean way that makes removing unwanted targets something extremely easy to do.

The benefits of being able to build `mold` from scratch with "native" optimizations in just 30 seconds in a system that has no access to a pre-packaged version, or one where that packaged version is too old, in my opinion, clearly outweighs the idea of hoping that any `mold` binary marked as version x.y.z will support the same architectures. Even worse - it force us to use forks or patched versions that can easily get out of date and miss security fixes.

In conclusion, please give it a second thought. And, regardless of what you decide to do, thanks again for the linker.

**Comment by rui314** [comment]
By popular request, I added a flag to specify the set of target architectures. As I wrote in the comment for the flag, please don’t abuse it. Thanks.
